src/main/java/thaumic/tinkerer/common/item/ItemCleansingTalisman.java

```diff
@@ -153,15 +163,34 @@ public void onWornTick(ItemStack par1ItemStack, EntityLivingBase player) {
         if (isEnabled(par1ItemStack) && !par2World.isRemote) {
             if (player.ticksExisted % 20 == 0) {
                 if (player instanceof EntityPlayer) {
-                    boolean removed = false;
-                    int damage = 1;
 
-                    Collection<PotionEffect> potions = player.getActivePotionEffects();
+                    // All effects we try to remove in this operation, including permanent effects.
+                    Collection<Integer> effectsToRemove = new ArrayList<>();
```

Comment:
Please use `Set` instead of `ArrayList` when you need to perform `contains` check often.

Reply:
Since we are only keeping track of very few debuffs at any time (super tank gives 4, I don't know of anything that gives more unless you deliberately combine different effects), my intuition was that a linear search of an array would be faster than an expensive logarithmic structure. But I haven't profiled this or anything. I will defer to your judgment -- I changed lists to HashMaps.